compiler/ast/types.nim

```diff
@@ -443,49 +449,29 @@ proc firstOrd*(conf: ConfigRef; t: PType): Int128 =
   of tyInt64: result = toInt128(0x8000000000000000'i64)
   of tyUInt..tyUInt64: result = Zero
   of tyEnum:
-    # if basetype <> nil then return firstOrd of basetype
-    if t.len > 0 and t[0] != nil:
-      result = firstOrd(conf, t[0])
-    else:
+    result =
       if t.n.len > 0:
         assert(t.n[0].kind == nkSym)
-        result = toInt128(t.n[0].sym.position)
+        toInt128(t.n[0].sym.position)
+      else:
+        # currently empty enums aren't support as a type, but they should be
+        Zero
   of tyGenericInst, tyDistinct, tyTypeDesc, tyAlias, tySink,
      tyStatic, tyInferred, tyUserTypeClasses, tyLent:
     result = firstOrd(conf, lastSon(t))
   of tyOrdinal:
-    if t.len > 0: result = firstOrd(conf, lastSon(t))
-    else:
-      conf.localReport InternalReport(
-        kind: rintUnreachable,
-        msg: "invalid kind for firstOrd(" & $t.kind & ')')
-
+    # TODO: convert to an invalid return and have callee handle it
```

Comment:
Did you meant `caller` instead of `callee` here?

Reply:
I absolutely do, thanks

Reply:
ultimately removed the comment as it should simply be a failure.